tests/IntegrationTest.php

```diff
@@ -37,7 +37,7 @@ protected static function getFixturesDir(): string
 
     protected static function getTempFile(): string
     {
-        return tempnam(sys_get_temp_dir(), 'PHP-CS-Fixer');
+        return sys_get_temp_dir().\DIRECTORY_SEPARATOR.'MyClass.php';
```

Comment:
why the change? it could be problematic when running different integration test in parallel

Reply:
How could it be problematic? Aren't the tests run in parallel already so it would crash already?

Also, isn't it a good practice not to have tests non-deterministic? If there is a need of having a different path for each test better would be to have a deterministic way to do it.

Reply:
I do not have good understanding of newest PHPUnit parallelization and I imagine that now/later it could make parallel test cases execution, not only test suites - in that case, if all test cases would run in parallel, they would fight and override same test file - that's why the test file was always different in original code.

would you mind to share the reasoning?
> why the change?

Reply:
Fixer `psr_autoloading` needs a file name to get a class name, with a name like `/tmp/PHP-CS-FixernJXNlg` that would not be possible: https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/v3.21.2/src/Fixer/Basic/PsrAutoloadingFixer.php#L106

Reply:
thanks for reasoning, now it makes more sense!